frontend/test/metabase/meta/Parameter.unit.spec.js

```diff
@@ -95,18 +100,169 @@ describe("metabase/meta/Parameter", () => {
   describe("stringParameterValueToMBQL", () => {
     describe("when given an array parameter value", () => {
       it("should flatten the array parameter values", () => {
-        expect(stringParameterValueToMBQL(["1", "2"], null)).toEqual([
-          "=",
-          null,
-          "1",
-          "2",
-        ]);
+        expect(
+          stringParameterValueToMBQL(
+            { type: "category/=", value: ["1", "2"] },
+            null,
+          ),
+        ).toEqual(["=", null, "1", "2"]);
       });
     });
 
     describe("when given a string parameter value", () => {
       it("should return the correct MBQL", () => {
-        expect(stringParameterValueToMBQL("1", null)).toEqual(["=", null, "1"]);
+        expect(
+          stringParameterValueToMBQL(
+            { type: "category/starts-with", value: "1" },
+            null,
+          ),
+        ).toEqual(["starts-with", null, "1"]);
+      });
+    });
+
+    it("should default the operator to `=`", () => {
+      expect(
+        stringParameterValueToMBQL(
+          { type: "category", value: ["1", "2"] },
+          null,
+        ),
+      ).toEqual(["=", null, "1", "2"]);
```

Comment:
This 100% should go in the shared query lib, to the point that I'd say it's even worth moving it over as part of this PR. (The other stuff I mentioned earlier would have been nice, but not so nice that it's worth redoing). Long-term, our goal is for the frontend JS code not deal with MBQL directly at all, but rather have all the interaction with it happen thru the shared lib. Porting all of `metabase-lib` (and other parts of the FE client that deal with MBQL, like this one, that aren't part of `metabase-lib`) to the shared lib is going to take some time, but we can at least make sure new code goes to the shared lib going forward so we don't make the project harder

Reply:
This code would also be 10x easier to write and test in Clojure

Reply:
If it makes you feel any better, these MBQL-related functions already existed--I added a bit of additional logic to them and also added unit tests since there were none before.

`stringParameterValueToMBQL` and `numberParameterValueToMBQL` now depend on code in `schema_metadata.js` so we'd need to port a bit of that as well. It'd add a decent amount of bulk to this PR.

I'm not against porting but I'd definitely need some help as the cljs side of things is still all new to me.